### ISSUE TITLE
[RESEARCH] imitation learning research report

### ISSUE THREAD

**Issue by ll7** [ll7/robot_sf_ll7#270]
### 1. **Add Automated Results Collection & Analysis**
Currently, the pipeline only *runs* the experiments but doesn't automatically collect, analyze, and report results in a research-ready format.

**Add:**
- Automatic collection of training metrics (convergence timesteps, sample efficiency)
- Statistical comparison between baseline and pre-trained approaches
- Automated figure generation (learning curves, performance distributions)
- Structured output following the training_summary.schema.json contract

### 2. **Create Research Report Generation Module**
Add a dedicated reporting module that generates publication-ready outputs:

**Add:**
- Markdown report generator with tables, figures, and statistical tests
- LaTeX export for paper integration (following the figure guidelines in dev_guide.md)
- Automatic hypothesis testing (e.g., "BC pre-training reduces timesteps by ≥30%")
- Reproducibility metadata (seeds, configs, git hash, environment)

### 3. **Implement Experimental Design Features**
Make the pipeline support proper experimental methodology:

**Add:**
- Multiple random seeds per condition (already partially there)
- Confidence intervals and statistical significance tests
- Ablation study support (e.g., vary BC epochs, dataset size)
- Automated hyperparameter tracking

### 4. **Add Metrics Dashboard & Visualization**
Create comprehensive analysis tools:

**Add:**
- Learning curve comparison (baseline vs pre-trained)
- Sample efficiency plot (timesteps to convergence)
- Success rate / collision rate over time
- Distribution plots for final performance

### 5. **Structure Output as Research Artifact**
Organize outputs to match research standards:

**Create:**
```
output/research_reports/
├── imitation_<timestamp>/
│   ├── report.md              # Main report
│   ├── report.pdf             # LaTeX-compiled version
│   ├── figures/               # All plots (PDF + PNG)
│   ├── data/                  # Raw metrics (JSON/CSV)
│   ├── metadata.json          # Reproducibility info
│   └── configs/               # Exact configs used
```

### 6. **Add Research-Specific CLI Arguments**

```python
--experiment-name          # Name for the research run
--num-seeds               # Number of random seeds
--hypothesis              # What you're testing
--generate-report         # Auto-generate markdown/LaTeX
--significance-level      # For statistical tests (default: 0.05)
--export-latex            # Generate publication figures
```

### 7. **Implement Comparison Engine**
Enhance the comparison step to be research-grade:

**Add:**
- Paired t-tests or Mann-Whitney U tests
- Effect size calculations (Cohen's d)
- Bootstrap confidence intervals (already available in benchmark code)
- Publication-ready comparison tables

### 8. **Add Reproducibility Tracking**
Ensure every report is fully reproducible:

**Track:**
- Git commit hash, branch, and diff status
- Exact Python package versions (`uv export`)
- Hardware specs (CPU, GPU, memory)
- Random seeds for all stages
- Timing information (wall-clock, CPU time)

### 9. **Create Report Template**
Add a structured report that auto-populates:

```markdown
# Imitation Learning for Robot Navigation: Experimental Report

## Abstract
[Auto-generated summary of results]

## Experimental Setup
- Environment: [config details]
- Baseline: PPO from scratch
- Treatment: BC pre-training + PPO fine-tuning
- Metrics: Sample efficiency, final performance
- Seeds: [list of seeds]

## Results
### Sample Efficiency
[Table + figure showing timesteps to convergence]

### Final Performance  
[Table + figure showing success/collision rates]

### Statistical Analysis
[Significance tests with p-values]

## Conclusion
[Auto-generated based on hypothesis testing]

## Reproducibility
[Git hash, commands, configs]
```

### 10. **Integration with Existing Tools**
Leverage what's already there:

- Use the **run tracker** (already integrated!) to capture detailed telemetry
- Use **benchmark aggregation** tools for confidence intervals
- Use existing **figure generation helpers** from dev_guide.md
- Export to **TensorBoard** for interactive exploration

---

## Concrete Implementation Steps

Would you like me to:

1. **Create a research report generation module** (`robot_sf/research/report_generator.py`)
2. **Add automated analysis scripts** that process tracker manifests into research outputs
3. **Create LaTeX templates** for publication-ready figures
4. **Implement statistical comparison tools** with hypothesis testing
5. **Add example research configuration** showing multi-seed experimental design
6. **Write documentation** on conducting research experiments with the pipeline

The most impactful first step would be **#1 + #2**: creating a report generator that consumes the tracker manifests (which you're already generating!) and produces a structured research report with figures and statistical analysis.
